### ISSUE TITLE
Consumer (Consumer<Flux<T>>) is not working with Flux type in Spring Cloud Stream 

### ISSUE THREAD

**Issue by msshitole** [spring-cloud/spring-cloud-function#237]
It is throwing an ClassCastException and Consumer fails to start it. Here is short stackTrace:
`org.springframework.cloud.stream.binder.BinderException: Exception thrown while starting consumer:
Caused by: java.lang.ClassCastException: com.prime.dataflow.reactivesinkdemo.ReactiveStreamSinkApplication$$Lambda$201/728258269 cannot be cast to java.util.function.Function
`
Here is the example code which causes above error:

`public static void main(String[] args) {
        SpringApplication.run(ReactiveStreamSinkApplication.class,
        "--spring.cloud.stream.function.definition=myconsumer");
    }

    @Bean
    public Consumer<Flux<String>> myconsumer() {
        return stream -> stream.subscribe(value -> {
            if ("foo".equals(value)) {
                throw new RuntimeException("BOOM!");
            }
            System.out.println("Received value: " + value);
        });
    }
`

Whereas if I remove Flux from the return type then Consumer (java.util.function.Consumer) works fine. Working code example:
`@Bean
    public Consumer<String> myconsumer() {
        return value -> {
            if ("foo".equalsIgnoreCase(value)) {
                throw new RuntimeException("BOOM!");
            }
            System.out.println("Received value: " + value);
        };
    }`

Basically as a user,  Its very handy to work with Flux and do some declarative style programming in myConsumer method. Whereas **Flux** works perfectly fine with the java.util.function.Supplier and java.util.function.Function. And It's very handy to work with Flux.

Just for a reference,  Here is the sample working Flux code with Supplier and Consumer.
`@Bean
    public Supplier<Flux<String>> emit() {
        return () -> Flux.interval(Duration.ofMillis(30000l))
                .map(aLong -> aLong + " : Hello msg");
    }
@Bean
    public Function<String, String> toUpperCase() {
        return s -> s.toUpperCase();
    }
`

Same way, I am expecting Flux must work with java.util.function.Consumer as well. Please check. Many thanks!

**Comment by olegz** [comment]
This issue is actually a duplicate to https://github.com/spring-cloud/spring-cloud-stream/issues/1539.
And while fixing it may very well require an additional feature in Spring Cloud Function, at the moment and for clarity the actual exception is rooted in how Spring Cloud Stream(SCSt) wiring the functions and thus is an SCSt issue, so I am marking it as duplicate, linking and closing.

**Comment by msshitole** [comment]
@olegz Thanks for confirming the issue and describing the cause and reason for the same. We will be looking forward to seeing the resolution. This feature will be very important for us to capitalize on the goodness of the SCF programming model.

**Comment by olegz** [comment]
Cool, as you can see it is slated to be in before the release so you'll have it in few weeks regardless.
